action/protocol/staking/bucket_pool.go

```diff
@@ -128,13 +130,27 @@ func NewBucketPool(sr protocol.StateReader) (*BucketPool, error) {
 }
 
 // Total returns the total amount staked in bucket pool
-func (bp *BucketPool) Total() *big.Int {
-	return new(big.Int).Set(bp.total.amount)
+func (bp *BucketPool) Total(sr protocol.StateReader) (*big.Int, uint64, error) {
+	bp.lock.RLock()
+	defer bp.lock.RUnlock()
+
+	height, err := sr.Height()
+	if err != nil {
+		return nil, 0, err
+	}
```

Comment:
what's the value of reading height from sr here? unify api?

Reply:
pls check latest commit